great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py

```diff
@@ -148,6 +157,164 @@ def get_batch_id(
 
         return batch_ids[0]
 
+    def get_metric(
+        self,
+        batch_id: str,
+        validator: Validator,
+        metric_name: str,
+        metric_domain_kwargs: Optional[dict] = None,
+        metric_value_kwargs: Optional[dict] = None,
+        enforce_numeric: Optional[bool] = False,
+        fill_nan_with_zero: Optional[bool] = True,
+        domain: Optional[Domain] = None,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+    ) -> Dict[
+        str,
+        Union[
+            Union[int, np.int32, np.int64, float, np.float32, np.float64],
+            Dict[str, Any],
+        ],
+    ]:
+        metric_domain_kwargs = build_metric_domain_kwargs(
+            batch_id=batch_id,
+            metric_domain_kwargs=metric_domain_kwargs,
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
+        # Obtain value kwargs from rule state (i.e., variables and parameters); from instance variable otherwise.
+        metric_value_kwargs: Optional[
+            dict
+        ] = get_parameter_value_and_validate_return_type(
+            domain=domain,
+            parameter_reference=metric_value_kwargs,
+            expected_return_type=None,
+            variables=variables,
+            parameters=parameters,
+        )
+        metric_configuration_arguments: Dict[str, Any] = {
+            "metric_name": metric_name,
+            "metric_domain_kwargs": metric_domain_kwargs,
+            "metric_value_kwargs": metric_value_kwargs,
+            "metric_dependencies": None,
+        }
+        metric_value: Union[
+            int, np.int32, np.int64, float, np.float32, np.float64
+        ] = validator.get_metric(
+            metric=MetricConfiguration(**metric_configuration_arguments)
+        )
+        if enforce_numeric:
+            if not is_numeric(value=metric_value):
+                raise ge_exceptions.ProfilerExecutionError(
+                    message=f"""Applicability of {self.__class__.__name__} is restricted to numeric-valued metrics \
+(value of type "{str(type(metric_value))}" was computed).
+"""
+                )
+            if np.isnan(metric_value):
+                if not fill_nan_with_zero:
+                    raise ValueError(
+                        f"""Computation of metric "{metric_name}" resulted in NaN ("not a number") value.
+"""
+                    )
+                metric_value = 0.0
+
+        return {
+            "value": metric_value,
+            "details": {
+                "metric_configuration": metric_configuration_arguments,
+            },
+        }
+
+    def get_metrics(
```

Comment:
I know we briefly discussed yesterday but if this method is only used for multibatch parameter builders, does it make sense to allow all parameter builders to have access to it?

My original thought was an additional ABC `MultiBatchParameterBuilder` that extends `ParameterBuilder` but I could see that adding an additional layer of complexity we might not want. Maybe a docstring to clarify the usage would be helpful?

Reply:
The line between single-batch and multi-batch will soon get very blurred as use cases will progressively weigh in favor of multi-batch, while single-batch will be a special case of multi-batch.

Reply:
Sounds good, thanks for the clarification! I'll try and update my mental model for the future.